### PR TITLE
Add additional tests for find_annotation_position_inside_polygon

### PR DESCRIPTION
**Pull request by ShivamRajSri**
### What is this PR

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other

Additional unit tests for `find_annotation_position_inside_polygon`.

### Why is this PR needed?

The function `find_annotation_position_inside_polygon` contains several edge-case behaviors described in the docstring, such as:

- handling polygons with insufficient vertices
- working with concave polygons
- repairing invalid/self-intersecting polygons using `buffer(0)`
- resolving geometries that may become `MultiPolygon`

Currently these behaviors were not fully tested. Adding these tests improves reliability and helps prevent regressions in future changes.

### What does this PR do?

This PR adds additional unit tests in:

tests/test_unit/test_find_annotation_position_inside_polygon.py

The new tests cover:

- polygons with fewer than 4 vertices returning `None`
- a simple convex polygon where the annotation point is near the expected center
- concave polygons where the centroid lies outside the geometry
- self-intersecting polygons that are repaired using `buffer(0)`
- verification that the returned annotation point lies inside the repaired geometry

These tests improve coverage of the polygon annotation helper function and validate behavior described in the function’s documentation.

---

## References

No related issue.  
Tests were added to improve coverage and validate documented edge cases.

---

## How has this PR been tested?

The new tests were executed locally using:

pytest tests/test_unit/test_find_annotation_position_inside_polygon.py -v

All tests pass locally. Existing tests were also run to ensure that no current functionality is affected.

Pre-commit hooks were run to ensure formatting and linting compliance.

---

## Is this a breaking change?

No.

This PR only adds tests and does not modify any existing functionality.

---

## Does this PR require an update to the documentation?

No.

This PR only adds unit tests and does not introduce any new functionality or change existing behavior.

---

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with pre-commit

### CODE REVIEW

File: tests/test_unit/test_find_annotation_position_inside_polygon.py [hunk 1]
```diff
@@ -66,3 +66,58 @@ def test_handles_invalid_polygons(invalid_vertices):
...
+        ]
+    )
+
+    result = find_annotation_position_inside_polygon(vertices)
+
+    assert result is not None
+    x, y = result
+
+    assert pytest.approx(x, abs=0.2) == 5
+    assert pytest.approx(y, abs=0.2) == 5
```
Comment (alessandrofelder) [inline tests/test_unit/test_find_annotation_position_inside_polygon.py]:
```suggestion
    assert pytest.approx(x, abs=0.01) == 5
    assert pytest.approx(y, abs=0.01) == 5
```
Based on local testing, we can make this tolerance a lot stricter.

**Review by alessandrofelder — APPROVED** [review]
Thanks for this @ShivamRajSri - many apologies for taking so long to get back to you!

The PR is small, self-contained and easy-to-read 🎉 

I have suggested a tiny change but I think this is ready to merge.
